pkg/operator/ceph/object/objectstore.go

```diff
@@ -704,3 +707,84 @@ func GetObjectBucketProvisioner(c *clusterd.Context, namespace string) string {
 	}
 	return provName
 }
+
+// CheckDashboardUser returns true if the user is configure else return false
+func checkDashboardUser(context *Context) (bool, error) {
+	args := []string{"dashboard", "get-rgw-api-access-key"}
+	cephCmd := ceph.NewCephCommand(context.Context, context.clusterInfo, args)
+	out, err := cephCmd.Run()
+
+	if string(out) != "" {
+		return true, err
+	}
+
+	return false, err
+}
+
+func enableRGWDashboard(context *Context) error {
+	checkDashboard, err := checkDashboardUser(context)
+	if err != nil {
+		logger.Debug("Unable to fetch dashboard user key for RGW, hence skipping")
+		return nil
+	}
+	if checkDashboard {
+		logger.Debug("Dashboard is already enabled")
+		return nil
+	}
+	user := ObjectUser{
+		UserID:      DashboardUser,
+		DisplayName: &DashboardUser,
+		SystemUser:  true,
+	}
+	u, errCode, err := CreateUser(context, user)
+	if err != nil || errCode != 0 {
+		return errors.Wrapf(err, "failed to create %q user", DashboardUser)
+	}
+	args := []string{"dashboard", "set-rgw-api-access-key", *u.AccessKey}
+	cephCmd := ceph.NewCephCommand(context.Context, context.clusterInfo, args)
+	_, err = cephCmd.Run()
+	if err != nil {
+		return errors.Wrapf(err, "failed to set user accesskey %q", DashboardUser)
+	}
+
+	args = []string{"dashboard", "set-rgw-api-secret-key", *u.SecretKey}
+	cephCmd = ceph.NewCephCommand(context.Context, context.clusterInfo, args)
+	_, err = cephCmd.Run()
+	if err != nil {
+		return errors.Wrapf(err, "failed to set user secretkey %q", DashboardUser)
+	}
+
+	return nil
+}
+
+func disableRGWDashboard(context *Context) error {
+	checkDashboard, err := checkDashboardUser(context)
+	if err != nil {
+		logger.Debug("Unable to fetch dashboard user key for RGW, hence skipping")
```

Comment:
Why is this not treated as an error? The calling function will report it as a warning anyway.

Also, please add the context that the user needs to understand what has gone wrong. What is being skipped (disabling the dashboard)? What user?
```suggestion
		return errors.Newf("unable to fetch dashboard user key for user %q", <user>)
```

Reply:
@BlaineEXE If the dashboard module is disabled, then it will return an error. We need to bypass that situation